photon-ml/src/main/scala/com/linkedin/photon/ml/Driver.scala

```diff
@@ -169,19 +169,26 @@ protected[ml] class Driver(
   }
 
   protected def prepareTrainingData(): Unit = {
+    // Verify that selected features files exist
     params.selectedFeaturesFile.foreach(file => {
       val path = new Path(file)
       if (! path.getFileSystem(sc.hadoopConfiguration).exists(path)) {
         throw new IOException("Could not find [" + file + "]. Check that the file exists")
       }
     })
+
+    // Read training data
     trainingData = suite.readLabeledPointsFromAvro(
       sc, params.trainDir, params.selectedFeaturesFile, params.minNumPartitions)
       .persist(trainDataStorageLevel).setName("training data")
-    featureNum = trainingData.first().features.size
+
     trainingDataNum = trainingData.count().toInt
+    require(trainingDataNum > 0,
+      "No training data found. Ensure that training data exists and feature vectors are not empty.")
```

Comment:
After a second look at ```GLMSuite```, I realized that the problem is caused by the fact that the created data will be empty if the features size is 0. So this error message actually reflects how we deal with data set with empty features in ```GLMSuite```.

I didn't know why GLMSuite was designed in that way, and will open an issue to separate logic of empty feature and empty data set checks.